Metro notifications

ethomson opened the pull request:
Introduce Metro notifications (Win 8 only).  Note that to display a metro notifications, you need to have a link installed on the start menu.  This will install an invisible link when the `debugInstall` configuration value is `true`.

Commit:
Add a dummy client for debugging

Commit:
Introduce MetroNotification

Commit:
Refactor tracing into Radar

ethomson commented on Radar/Notifications/ConsoleNotification.cs:
```diff
@@ -30,8 +31,12 @@ public void Notify(Client client, Event e)
-                client.Name, e.Time, e.Identity.Name, e.Identity.Email, e.RepositoryFriendlyName, e.Kind, e.BranchName, string.Join(", ", e.Shas));
+            string content = e.Content ??
+                String.Format("{0} {1} {2} [{3}]",
+                e.RepositoryFriendlyName, e.Kind, e.BranchName, string.Join(", ", e.Shas));
```
This is terrible.  Let's fix Events.

nulltoken replied:
Actually I thought about generating the content from `Event`, but I realized that if you were willing to introduce links or typographical enrichments in the notifications, you may need the pieces of information, rather than the dumped string. Or was I completely wrong?

ethomson replied:
Ugh, I don't know.  Right now we have two notifiers (console and metro) that can only deal with a string.  (Though the metro notifier _can_ do something when clicked, which may be interesting.)  But neither can display anything more rich than that.

I realize the console one is very stupid and not particularly worth mentioning in this argument.  But if we're going to support multiple notifications, I don't want to have some complex system that can understand every type of message that every type of client could possibly send.

There's already a lot and I could easily envision some GitHub and VSO specific clients that can notify you about CI builds from recent pushes, work items created / assigned to you, test failures, etc.  I would very much hate to code this message-creation process in every notification system.

I think that the client knows what it wants to tell the end-user.

We could make Event an interface, and then let the events know how to produce the messages; or we could have Clients produce the messages...  We could have a `Content` property (or a `GetContent` function) and if we want to include links, we could do a `RichContent` that produces HTML or something...

nulltoken commented on Radar/Radar.csproj:
```diff
@@ -39,22 +40,30 @@
     <Reference Include="Microsoft.CSharp" />
     <Reference Include="System.Data" />
     <Reference Include="System.Xml" />
+    <Reference Include="Windows" />
```
This seems to be missing on my Win8.1 box? Should we install a specific dependency? If it isn't available through NuGet, maybe an addition to the README file would make sense?

ethomson replied:
Hmm.  It should be.  This is the new Windows RT API.  When you set TargetPlatformVersion=8 you should be able to take a reference on this (though you won't find this DLL on your system).  There's much magic going on here behind the scenes and I don't know all the details, but it should work in Windows 8.0 or better and VS 2013 or better.

Which VS are you running that this is failing?

nulltoken committed:
Fix references

nulltoken committed:
Protect RecentsEvents from multithreaded access

nulltoken committed:
Prefer a ConcurrentDictionary over a plain Dictionary

nulltoken committed:
Copy config.sample to output directory

nulltoken committed:
Wait for complete intialization

nulltoken committed:
Silently disregard remote tags for now

nulltoken committed:
Minor renamings

nulltoken committed:
Introduce NullIdentity type

 - Used when the identity cannot be inferred from the action

nulltoken committed:
Retrieve identity from topmost commit

nulltoken committed:
Minor refactoring

nulltoken committed:
Build Event.Content from BranchEvent

 - Remove now useless Event properties
 - Make ConsoleNotification format the message differently whether the identity is known or not

nulltoken committed:
Improve retrieval of commits on created branches

ethomson committed:
Merge pull request #4 from ethomson/ntk/fixups

Some fixups

Commit:
Revert "Fix references"

This reverts commit d996a89dd132c0f3508f57c31b695c0de9a89bf1.

nulltoken committed:
Remove Radar dependency from RemoteRepositoryTracker

nulltoken committed:
Add basic MonitoredRepository tests

nulltoken committed:
Add basic RemoteRepositoryTracker test

nulltoken committed:
Introduce IEvent

nulltoken committed:
Make Radar.Util.Assert internal

nulltoken committed:
Test detection of BranchCreatedFromAKnownCommit

nulltoken committed:
Fix typo

nulltoken committed:
Test detection of BranchResetToAKnownCommit

nulltoken committed:
Test detection of BranchDeleted

nulltoken committed:
Test detection of BranchCreated

nulltoken committed:
Test detection of BranchUpdated

nulltoken committed:
Test detection of BranchForceUpdated

nulltoken committed:
Minor refactoring

nulltoken committed:
Fix detection of creation of a new root branch

nulltoken committed:
Introduce DummyEvent

nulltoken committed:
Rename Event to RepositoryEvent

ethomson committed:
Merge pull request #10 from ethomson/ntk/tests

TESTS!!!!!!!!!!!!!!!
